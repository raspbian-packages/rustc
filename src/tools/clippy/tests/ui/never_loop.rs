#![feature(plugin)]
#![plugin(clippy)]
#![allow(single_match, unused_assignments, unused_variables)]

fn test1() {
    let mut x = 0;
    loop { // never_loop
        x += 1;
        if x == 1 {
            return
        }
        break;
    }
}

fn test2() {
    let mut x = 0;
    loop {
        x += 1;
        if x == 1 {
            break
        }
    }
}

fn test3() {
    let mut x = 0;
    loop { // never loops
        x += 1;
        break
    }
}

fn test4() {
    let mut x = 1;
    loop {
        x += 1;
        match x {
            5 => return,
            _ => (),
        }
    }
}

fn test5() {
    let i = 0;
	loop { // never loops
        while i == 0 { // never loops
            break
        }
        return
	}
}

fn test6() {
    let mut x = 0;
    'outer: loop { // never loops
        x += 1;
		loop { // never loops
            if x == 5 { break }
			continue 'outer
		}
		return
	}
}

fn test7() {
    let mut x = 0;
    loop {
        x += 1;
        match x {
            1 => continue,
            _ => (),
        }
        return
    }
}

fn test8() {
    let mut x = 0;
    loop {
        x += 1;
        match x {
            5 => return,
            _ => continue,
        }
    }
}

fn test9() {
    let x = Some(1);
    while let Some(y) = x { // never loops
        return
    }
}

fn test10() {
    for x in 0..10 { // never loops
        match x {
            1 => break,
            _ => return,
        }
    }
}

fn test11<F: FnMut() -> i32>(mut f: F) {
    loop {
        return match f() {
            1 => continue,
            _ => (),
        }
    }
}

fn main() {
    test1();
    test2();
    test3();
    test4();
    test5();
    test6();
    test7();
    test8();
    test9();
    test10();
    test11(|| 0);
}

