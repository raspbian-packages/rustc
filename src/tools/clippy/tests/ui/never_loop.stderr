error: this loop never actually loops
  --> $DIR/never_loop.rs:7:5
   |
7  | /     loop { // never_loop
8  | |         x += 1;
9  | |         if x == 1 {
10 | |             return
11 | |         }
12 | |         break;
13 | |     }
   | |_____^
   |
   = note: `-D never-loop` implied by `-D warnings`

error: this loop never actually loops
  --> $DIR/never_loop.rs:28:5
   |
28 | /     loop { // never loops
29 | |         x += 1;
30 | |         break
31 | |     }
   | |_____^

error: this loop never actually loops
  --> $DIR/never_loop.rs:47:2
   |
47 | / \tloop { // never loops
48 | | \t       while i == 0 { // never loops
49 | | \t           break
50 | | \t       }
51 | | \t       return
52 | | \t}
   | |__^

error: this loop never actually loops
  --> $DIR/never_loop.rs:48:9
   |
48 | /         while i == 0 { // never loops
49 | |             break
50 | |         }
   | |_________^

error: this loop never actually loops
  --> $DIR/never_loop.rs:57:5
   |
57 | /     'outer: loop { // never loops
58 | |         x += 1;
59 | | \t\tloop { // never loops
60 | |             if x == 5 { break }
...  |
63 | | \t\treturn
64 | | \t}
   | |__^

error: this loop never actually loops
  --> $DIR/never_loop.rs:59:3
   |
59 | / \t\tloop { // never loops
60 | | \t\t          if x == 5 { break }
61 | | \t\t\tcontinue 'outer
62 | | \t\t}
   | |___^

error: this loop never actually loops
  --> $DIR/never_loop.rs:92:5
   |
92 | /     while let Some(y) = x { // never loops
93 | |         return
94 | |     }
   | |_____^

error: this loop never actually loops
   --> $DIR/never_loop.rs:98:5
    |
98  | /     for x in 0..10 { // never loops
99  | |         match x {
100 | |             1 => break,
101 | |             _ => return,
102 | |         }
103 | |     }
    | |_____^

error: aborting due to 8 previous errors

